File: services/mm_event/redis/keys.go
package redis

import "fmt"

type playerKeys struct {
	id int64
}

func (ks playerKeys) rating() string {
	return fmt.Sprintf("rating:%d", ks.id)
}

func (ks playerKeys) fleet() string {
	return fmt.Sprintf("fleet:%d", ks.id)
}

func (ks playerKeys) name() string {
	return fmt.Sprintf("name:%d", ks.id)
}

func (ks playerKeys) displayedRating() string {
	return fmt.Sprintf("displayed_rating:%d", ks.id)
}

func (ks playerKeys) regId() string {
	return fmt.Sprintf("reg_id:%d", ks.id)
}

func (ks playerKeys) keys() []string {
	return []string{ks.rating(), ks.fleet(), ks.name(), ks.displayedRating(), ks.regId()}
}

File: services/mm_exec/redis_repo/keys.py
from dataclasses import dataclass


@dataclass
class PlayerKeySet:
    id: str | int

    @property
    def rating(self):
        return f"rating:{self.id}"

    @property
    def fleet(self):
        return f"fleet:{self.id}"

    @property
    def name(self):
        return f"name:{self.id}"

    @property
    def displayed_rating(self):
        return f"displayed_rating:{self.id}"

    @property
    def reg_id(self):
        return f"reg_id:{self.id}"

    def keys(self):
        return [self.rating, self.fleet, self.name, self.displayed_rating, self.reg_id]
